src/python/pants/jvm/test/junit.py

```diff
@@ -65,17 +69,26 @@ async def setup_junit_for_target(
     test_subsystem: TestSubsystem,
 ) -> TestSetup:
 
-    jdk = await Get(
-        JdkEnvironment, JdkRequest, JdkRequest.from_field(request.field_set.jdk_version)
+    jdk, dependencies = await MultiGet(
+        Get(JdkEnvironment, JdkRequest, JdkRequest.from_field(request.field_set.jdk_version)),
+        Get(Targets, DependenciesRequest(request.field_set.dependencies)),
```

Comment:
(applies to both implementations)

One thing to think about here is whether this should be for direct or transitive dependencies. In `v1` for JVM it was transitive, which has the advantage of being able to declare test helpers in other targets with their own `files` target dependencies (a test helper using a `yaml` file, for example). I know less about the semantics of Go though, and whether it would be useful (or even possible) to be transitive in that case cc @Eric-Arellano , @tdyas.